athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -261,12 +312,37 @@ public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest
     {
         logger.info("doGetSplits: enter - " + request);
 
-        //Every split must have a unique location if we wish to spill to avoid failures
+        String domain = request.getTableName().getSchemaName();
+        String endpoint = getDomainEndpoint(domain);
+
+        // Every split must have a unique location if we wish to spill to avoid failures
         SpillLocation spillLocation = makeSpillLocation(request);
 
-        //Since our connector does not support parallel reads we return a fixed split.
-        return new GetSplitsResponse(request.getCatalogName(),
-                Split.newBuilder(spillLocation, makeEncryptionKey()).build());
+        // Create split
+        Split.Builder splitBuilder = Split.newBuilder(spillLocation, makeEncryptionKey());
+        if (endpoint != null) {
+            // Add domain and endpoint to the split to be used by the Record Handler.
+            splitBuilder.add(domain, endpoint);
+        }
+
+        return new GetSplitsResponse(request.getCatalogName(), splitBuilder.build());
+    }
+
+    /**
+     * Gets an endpoint from the domain mapping. For AWS Elasticsearch Service, if the domain does not exist in
+     * the domain map, refresh the latter by calling the AWS ES SDK (it's possible that the domain was added
+     * after the last connector refresh).
+     * @param domain is used for searching the domain map for the corresponding endpoint.
+     * @return endpoint corresponding to the domain or a null if domain does not exist in the map.
+     */
+    private String getDomainEndpoint(String domain)
+    {
+        if (!domainMap.containsKey(domain) && helper.isAutoDiscoverEndpoint()) {
+            logger.warn("Unable to find domain in map! Attempting to refresh map...");
```

Comment:
which domain were we unable to find? might be useful to log it.

Also at the end you should check if the result is null, if so... throw a useful error message instead of returning null.

Reply:
Added domain to log message as well as an additional check for null endpoint with a thrown exception.